src/odemis/driver/actuator.py

```diff
@@ -1882,39 +1889,54 @@ def _doMoveAbs(self, pos):
         position = position + self._metadata[model.MD_POS_COR]
         logging.debug("Moving axis %s (-> %s) to %g", self._axis, self._caxis, position)
 
-        # do referencing after i=5 moves or when actuator has accumulated 2pi steps
-        # --> do correct for accumulated errors
-        if self._move_num_total == 5 or abs(self._move_sum) >= self._cycle:
-            # to get rid of accumulated error
+        self._move_num_total += 1
 
-            self._child.moveRel({self._caxis: self._offset}).result()
+        # calc move needed to reach requested pos
+        move, cur_pos = self._findShortestMove(position)
+
+        # Check that the move passes by the reference switch by detecting whether
+        # the current and final position are not in the same multiple of cycle.
+        # "Linear" view of the axis (c = cycle)
+        #    -2c       -1c        0         c         2c        3c
+        # ----|---------|---------|---------|---------|---------|---
+        #         -2            -1      0        1        2
+        final_pos = cur_pos + move
+        pass_ref = (cur_pos // self._cycle) != (final_pos // self._cycle)
+        
+        if pass_ref or self._move_num_total >= 5:
+
+            # Move to pos close to ref switch
+            move, cur_pos = self._findShortestMove(position)
+                
+            self._child.moveRel({self._caxis: move}).result()
             self._child.reference({self._caxis}).result()
 
-            self._move_sum = 0
+            # now calc how to move to the actual position requested
+            move, cur_pos = self._findShortestMove(position)
+                
             self._move_num_total = 0
 
-        # Optimize by moving through the closest way
+        _child_future = self._child.moveRel({self._caxis: move})
+        _child_future.result()
+
+        # TODO: needed when cancel will be implemented
+        # self._child_future = self._child.moveRel({self._caxis: move})
+        # self._child_future.result()
+        # self._child_future = None
+
+    def _findShortestMove(self, position):
```

Comment:
Could you rename `position` to something like `tpos` or `target_pos`, to avoid confusion between all the different "positions"?